src/libraries/System.Text.RegularExpressions/tests/UnitTests/RegexReductionTests.cs

```diff
@@ -580,9 +582,14 @@ public void PatternsReduceIdentically(string actual, string expected)
         [InlineData("(?=a(b*)c)", "(?=ab*c)")]
         [InlineData("(?=a((((b))))c)", "(?=abc)")]
         [InlineData(@"a*(?=a)", @"(?>a*)(?=a)")]
-        [InlineData(@"a*(?!b)b", @"(?>a*)(?!b)b")]
+        [InlineData(@"a*(?!b)b", @"(?>a*)(?!a)b")]
         [InlineData(@"a*(?<!b)cde", @"(?>a*)(?<!b)cde")]
         [InlineData(@"a*(?<=b)cde", @"(?>a*)(?<=b)cde")]
+        [InlineData(@"a*(?=)a", @"(?>a*)(?=)a")]
+        [InlineData(@"(?<=(ab))", @"(?<=ab)")]
+        [InlineData(@"(?=(ab|ac))", @"(?=a[bc])")]
+        [InlineData(@"(?<=(ab|ac))", @"(?<=a[bc])")]
```

Comment:
I take it these are transformations that could be valid if the contents of the lookarounds were noncapturing groups instead?

Reply:
Some of them. `a*(?=)a` can't have the loop be made atomic because the lookahead is nullable. The rest could have the capture groups removed if they were negative lookarounds, but they're positive. These tests aren't strictly related to this change, but I had them and figured I'd include them.